flink-core/src/main/java/org/apache/flink/metrics/reporter/JMXReporter.java

```diff
@@ -73,10 +86,61 @@ public JMXReporter() {
 	// ------------------------------------------------------------------------
 
 	@Override
-	public void open(Configuration config) {}
+	public void open(Configuration config) {
+		this.jmxServer = startJmxServer(config);
+	}
+
+	private static JMXServer startJmxServer(Configuration config) {
+		JMXServer jmxServer;
+
+		String portRange = config.getString(KEY_METRICS_JMX_PORT, "9010-9025");
+		String[] ports = portRange.split("-");
+
+		if (ports.length == 0 || ports.length > 2) {
+			throw new IllegalArgumentException("JMX port range was configured incorrectly. " +
+				"Expected: <startPort>[-<endPort>] Configured: " + portRange);
+		}
+
+		if (ports.length == 1) { //single port was configured
```

Comment:
Have you checked out `NetUtils.getPortRangeFromString` to iterate over a range of ports? This returns an `Iterator` over all defined ports. With that you can also define port ranges like: `1911, 2222-3333, 42-1337`.